allen-cell-animated/nucmorph-colorizer
# refactor: Remove color ramps from UI (with backwards-compatibility)

Problem
=======
First part of #217. Adds the ability to hide color ramps and palettes from the UI while still allowing them to be loaded via the URL. This supports backwards-compatibility with old or outdated ramps.

*Estimated review time: small, 15 minutes* 

Solution
========
- Refactors the color ramps and categorical palettes to have a list of known data and a separate list of displayed keys.
- Refactors `ColorRampDropdown` to only show palettes in the list of displayed keys.
- Updates unit tests

## Type of change
* New feature (non-breaking change which adds functionality)

Steps to Verify:
----------------
1. Open the preview link and observe that a color map was loaded from the URL that does not appear in the list of options: https://allen-cell-animated.github.io/nucmorph-colorizer/pr-preview/pr-245/?dataset=https%3A%2F%2Fraw.githubusercontent.com%2Fallen-cell-animated%2Fnucmorph-colorizer%2Fmain%2Fdocumentation%2Fexample_dataset%2Fmanifest.json&feature=Continuous%20Feature&color=esri-orange_5&palette-key=iwanthue_pastel_3
3. Switch to a categorical feature. The displayed color palette should not match any in the dropdown.

Screenshots (optional):
-----------------------
*Notice that "ESRI Orange 5" is not in the list of options but is still being rendered!*

![image](https://github.com/allen-cell-animated/nucmorph-colorizer/assets/30200665/97afddbd-e1a8-4eea-b0fd-805e2ae0d13c)
